### PR TITLE
[wxwidgets] Fix cmake wrapper

### LINKED ISSUE #16543: [wxWidgets] Using static triplet fails because of missing png.lib
**Describe the bug**
The cmake wrapper for wxWidgets expects libpng to always be called `png.lib` but it is not on static triplets so it fails to find the library to link when you try to use wxWidgets.

**Environment**
- OS: Windows
- Compiler: cl.exe 19.28.29910

**To Reproduce**
1. ./vcpkg install wxwidgets:x64-windows-static
2. Try to use wxWidgets from your own statically linked project via CMake

**Expected behavior**
Successful linking

**Failure logs** 
    fatal error LNK1104: cannot open file 'png.lib'

**Additional context**
libpng's library for static triplets is called `png_static` so using `png` won't work.

I have a pull request ready that I will add after filing this issue.

### LINKED ISSUE #16545: [wxWidgets] static triplet only debug/release library mixup
**Describe the bug**
When using wxWidgets supplied by vcpkg through CMake in a statically linked debug configuration project, the linker gets supplied with the release versions of the wxwidgets libraries instead of the debug versions.

**Environment**
- OS: Windows
- Compiler: cl.exe 19.28.29910

**To Reproduce**
1. ./vcpkg install wxwidgets:x64-windows-static
2. Use wxWidgets in a statically linked debug project
3. See error

I've attached a basic [CMakeLists.txt](https://github.com/microsoft/vcpkg/files/6087507/CMakeLists.txt) and the [wxWidget's reference hello world](https://github.com/microsoft/vcpkg/files/6087510/foo.cpp.txt) to reproduce. I'm using VS 2019 CMake support and specifying my CMake toolchain and vcpkg triplet in my CMakeSettings.json.

**Expected behavior**
Debug libraries should be provided. This works correctly on x64-windows.

**Failure logs** 
[wxwidgets-static-debrel-mixup.txt](https://github.com/microsoft/vcpkg/files/6087500/wxwidgets-static-debrel-mixup.txt)

**Additional context**
Getting to this point relies on a PR I made to fix an earlier issue with linking libpng: #16544

### PR DESCRIPTION
**Pull request by JackBoosY**
Since the target of libpng in static is libpng_static, fix the usage issue.

Fixes #16543.
suspend #16544.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Will also fix #16545 in this PR.

**Comment by Hoikas** [comment]
I think it would be better to patch the `png` target to erase the difference between `png` and `png-static`.

**Comment by JackBoosY** [comment]
@Hoikas  I think we'd better don't do that because it's official.

File: versions/w-/wxwidgets.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "3b3723cdaaa7cd2020f4d35dd5566767d72fceff",
+      "version-semver": "3.1.4",
+      "port-version": 7
```
Comment (JackBoosY) [inline versions/w-/wxwidgets.json:6]:
```suggestion
      "port-version": 8
```

File: versions/w-/wxwidgets.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "3b3723cdaaa7cd2020f4d35dd5566767d72fceff",
```
Comment (JackBoosY) [inline versions/w-/wxwidgets.json:4]:
```suggestion
      "git-tree": "408ffb613207d8a5056d06b48e570b6f17f73855",
```

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/17277.

**Comment by PhoebeHui** [comment]
@Hoikas, Similar discussion in https://github.com/microsoft/vcpkg/issues/14930.

File: ports/wxwidgets/vcpkg-cmake-wrapper.cmake [hunk 3]
```diff
@@ -15,5 +15,11 @@ find_package(TIFF QUIET)
 find_package(expat CONFIG QUIET)
 
 if(wxWidgets_LIBRARIES AND NOT wxWidgets_LIBRARIES MATCHES "TIFF::TIFF;png;expat::expat;ZLIB::ZLIB")
-    list(APPEND wxWidgets_LIBRARIES "TIFF::TIFF;png;expat::expat;ZLIB::ZLIB")
-endif()
+    list(APPEND wxWidgets_LIBRARIES "TIFF::TIFF;expat::expat;ZLIB::ZLIB")
+
+    if (TARGET png)
+        list(APPEND wxWidgets_LIBRARIES "png")
+    else()
```
Comment (PhoebeHui) [inline ports/wxwidgets/vcpkg-cmake-wrapper.cmake:22]:
```suggestion
    elseif(TARGET png_static)
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: versions/w-/wxwidgets.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "408ffb613207d8a5056d06b48e570b6f17f73855",
```
Comment (JackBoosY) [inline versions/w-/wxwidgets.json:4]:
```suggestion
      "git-tree": "8fc2d91de4a0a86698c67e6bb662c14cbfd29732",
```

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**